lib/ddtrace/contrib/configuration/settings.rb

```diff
@@ -12,6 +12,9 @@ class Settings
         option :service_name
         option :tracer do |o|
           o.delegate_to { Datadog.tracer }
+          o.on_set do |_value|
```

Comment:
What's nice about this `on_set` is it'll always let us know if we accidentally set this when we shouldn't have in integration/test code.